### PR TITLE
Improve Hexagon power and performance APIs to reduce overhead.

### PR DESCRIPTION
**Pull request by dsharletg**
The old power_hvx_on and off functions also set the performance. This added a lot of extra overhead (~5 ms, now down to < 1 ms).

With these new APIs, it appeared to not even be necessary to power HVX on/off explicitly in HelloHexagon, which is a pipeline that runs in 2 ms, so the overhead must be very low.

### CODE REVIEW

**Comment by dsharletg** [comment]
There's still an open issue to figure out with this PR: old applications got the turbo setting for free. if these applications use a runtime from this PR or later, they won't get that anymore, and so performance will probably be bad. There might be a way to keep that default behavior, but I haven't figured out how yet. Setting performance to turbo in initialize_kernels doesn't work because it will wipe out the app's preference every time a new kernel is initialized.

**Comment by ronlieb** [comment]
LGTM

**Comment by dsharletg** [comment]
Two more issues:

- With the reduced overhead of power_hvx_on, it might not make sense to insert the power_hvx_on calls around pipelines anymore, because the overhead of the RPC call is likely higher than the overhead of powering HVX on/off redundantly. For cases where this isn't true, it means the pipelines running on Hexagon must be *tiny*, and the user can still add their own power_hvx_on/off calls around their multiple pipeline invocations.

- The latency setting probably needs to be modified for the low power modes. I added this, PTAL.

**Comment by ronlieb** [comment]
lgtm

File: src/runtime/hexagon_remote/halide_hexagon_remote.idl [hunk 1]
```diff
@@ -31,7 +31,18 @@ interface halide_hexagon_remote
...
+                         in unsigned long mipsPerThread,
+                         in unsigned long mipsTotal,
+                         in long set_bus_bw,
+                         in unsigned long bwMegabytesPerSec,
+                         in unsigned long busbwUsagePercentage,
+                         in long set_latency,
+                         in long latency);
+
+    // Deprecated APIs
     long power_hvx_on_mode(in long mode);
```
Comment (dpalermo) [inline src/runtime/hexagon_remote/halide_hexagon_remote.idl]: We would be OK if these two interfaces would be completely removed, not just deprecated.

Comment (dsharletg) [inline src/runtime/hexagon_remote/halide_hexagon_remote.idl]: Note that they are already removed from HalideRuntimeHexagonHost.h, these are just here to enable old apps to run on new runtimes. So basically, old apps won't *build* with this change, but they will run. Are you are still OK with removing these?

Comment (dpalermo) [inline src/runtime/hexagon_remote/halide_hexagon_remote.idl]: Yes.  We are OK with that behavior.  The power_hvx_on_mode/perf interfaces haven't been there that long, and the group that requested them will be fine with the new interfaces, and they are still just evaluating using Halide right now...so right now would be a good opportunity to remove them completely.

File: src/runtime/HalideRuntimeHexagonHost.h [hunk 2]
```diff
@@ -83,12 +86,12 @@ typedef struct {
...
+ * the performance levels they want, which may or may not be respected
+ * by Hexagon. */
 // @{
-extern int halide_hexagon_power_hvx_on(void *user_context);
-extern int halide_hexagon_power_hvx_on_mode(void *user_context, halide_hvx_power_mode_t mode);
-extern int halide_hexagon_power_hvx_on_perf(void *user_context, halide_hvx_power_perf_t *perf);
-extern int halide_hexagon_power_hvx_off(void *user_context);
-extern void halide_hexagon_power_hvx_off_as_destructor(void *user_context, void * /* obj */);
+extern int halide_hexagon_set_performance_mode(void *user_context, halide_hvx_power_mode_t mode);
+extern int halide_hexagon_set_performance(void *user_context, halide_hvx_power_perf_t *perf);
```
Comment (dpalermo) [inline src/runtime/HalideRuntimeHexagonHost.h]: Needs to be halide_hexagon_set_performance_perf...or the symbol in hexagon_host.cpp needs to be changed to halide_hexagon_set_performance (no preference)

**Comment by dsharletg** [comment]
Going to merge this, the last commit was just merging master, and none of this can affect travis anyways (except the convolution correctness test cleanup).

**Pull request merged**